ERP-BACKEND/app/services/search_service.py

```
@@ -58,32 +58,68 @@ def index_entity(self, entity_type: str, entity_id: int, title: str, content: st
+                    'searchable_text': stmt.excluded.searchable_text,
+                    'meta_data': stmt.excluded.meta_data,
+                    'tags': stmt.excluded.tags,
+                    'updated_at': datetime.utcnow()
+                }
+            )
+            self.db.execute(stmt)
             self.db.commit()
+        except IntegrityError as e:
+            self.db.rollback()
+            # Fallback to individual indexing for problematic records
+            logger = logging.getLogger(__name__)
```

Ruff (0.16.1)

[error] Undefined name `logging`

(F821)